cvat/apps/engine/views.py

```diff
@@ -887,23 +959,36 @@ def upload_finished(self, request):
             task_data = self._object.data
             serializer = DataSerializer(task_data, data=request.data)
             serializer.is_valid(raise_exception=True)
-            data = dict(serializer.validated_data.items())
-            uploaded_files = task_data.get_uploaded_files()
-            uploaded_files.extend(data.get('client_files'))
-            serializer.validated_data.update({'client_files': uploaded_files})
```

Comment:
I'm confused about how this is supposed to work now. Won't a `client_files` field in the `Upload-Finish` request now overwrite this field in the database, causing all previously-recorded files to be forgotten?

Reply:
No, the only thing being done in the serializer is that the new rows are added in the DB, so no overwrites happen. Now, we just record everything the user sent on reach request, and optionally sort the data fetched from the DB during the last request processing.